src/components/RouletteMain/RouletteMain.jsx

```diff
@@ -179,17 +185,40 @@ function RouletteMain(props) {
                     onCloseAction={onCloseAction}
                     onDeleteMovieConfirmAction={onDeleteMovieConfirmAction}
                 /> : null}
-        </>
+        </>)
+    }
+
+    return (
+        <Router>
+            <Switch>
+                <Route exact path="/" children={() => {
+                    navigateToMovieDetails(-1)
```

Comment:
All this logic looks too complicated. You just need to create separate component for movie list and use Route to render needed parts based on URL.